src/scripts/db.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/python
+# -*- coding: latin-1 -*-
+
+import mysql.connector # SQL library of choice 
+import secrets # Local file that contains config info 
+import re # regular expressions
+import adj # 133 adjectives 
+import test # test names 
+import random # random generation
+
+# create a connection pool 
+    # https://dev.mysql.com/doc/connector-python/en/connector-python-connection-pooling.html
+
+# pull down one thousand (?) records at a time 
+    # https://dev.mysql.com/doc/connector-python/en/connector-python-example-cursor-select.html
+
+# Script runs, like, 1345 times (L0L). Needs to save everything to local memory so that repeat codes are not created (?)
+
+# use the same pool (??) to post records to the other table
+    # https://dev.mysql.com/doc/connector-python/en/connector-python-example-cursor-transaction.html
+
+records = [] # Records from users_fb we need
+codes = [] # generated codes for user_referral_codes_v2
+# existingcodes = [] 
+ids = [] # auto increment id value from fb_users that is foreign key in user_referral_codes_v2
+set1 = set()# sets are much faster to look up in than lists. 
+set2 = set() # codes that already exist in user_referral_codes_v2
+
+
+return_value = "temp"
+
+pull = ""
+
+def genNewCode(code): # checks to see if generated code is a duplicate. pretty inefficient.
+    exists = True
+
+    while exists: # Loops until code is unique.  
+        if code not in set1 and code not in set2: 
+            exists = False
+        else:
+            last_slash = code.rfind('-')
+            num = code[last_slash-len(code) + 1:] # Save last character. 
+            code = code[:last_slash-len(code) + 1] # Splice that removes last few characters. 
+            num = int(num) + 1 # Increment character. 
+            code += str(num)
+
+    return code
+
+def pull_ids(): # pulls the ids from user_referral_codes_v2 and adds them to existingcodes so that we can check that they are not duplicates
+    print("Pulling existing ids...")
+    cnx = mysql.connector.connect(**secrets.read_replica)
+
+    cursor = cnx.cursor(buffered=True)
+
+    pull = "select v2_code from user_referral_codes_v2"
+
+    cursor.execute(pull)
+
+    for ids in cursor:
+        set2.add(ids[0]) 
+
+
+    print("Existing ids pulled!")
+
+def pull_info(): # pulls id and first_name from users_fb UNLESS the id exists in user_referral_codes_v2. Stores all in memory. 
+    print("Pulling info from users_fb")
+    cnx = mysql.connector.connect(**secrets.read_replica)
+
+    cursor = cnx.cursor(buffered=True)
+
+    min_id = 0
+    max_id = 1000 
+
+    while max_id < 100000 or return_value == '': # until empty string is returned. 
```

Comment:
is `max_id` meant to be the # of rows in `users_fb` table?  Wondering if this it's supposed to be set to something much higher like 1,348,000?

Reply:
It is ! Forgot to change it after testing.